libs/math/include/math/tensor_broadcast.hpp

```diff
@@ -175,5 +216,55 @@ inline bool Broadcast(F function, Tensor<T, C> &a, Tensor<T, C> &b, Tensor<T, C>
   return true;
 }
 
+template <typename F, typename T, typename C>
+inline bool Broadcast(F function, const Tensor<T, C> &a, const Tensor<T, C> &b, Tensor<T, C> &c)
```

Comment:
same as upgrade iterator - make a shared implementation method; however you will need two templates:

template <typename IteratorType1, typename IteratorType2>